parsers/EIA.py

```diff
@@ -361,133 +367,163 @@ def fetch_production_mix(zone_key, session=None, target_datetime=None, logger=No
         if len(correct_mix) > 0:
             correct_mixes.append(correct_mix)
 
-    return merge_production_outputs(correct_mixes, zone_key, merge_source='eia.gov')
+    return merge_production_outputs(correct_mixes, zone_key, merge_source="eia.gov")
 
 
-def fetch_exchange(zone_key1, zone_key2, session=None, target_datetime=None, logger=None):
-    sortedcodes = '->'.join(sorted([zone_key1, zone_key2]))
-    exchange = _fetch_series(sortedcodes, EXCHANGES[sortedcodes], session=session,
-                             target_datetime=target_datetime, logger=logger)
+def fetch_exchange(
+    zone_key1, zone_key2, session=None, target_datetime=None, logger=None
+):
+    sortedcodes = "->".join(sorted([zone_key1, zone_key2]))
+    exchange = _fetch_series(
+        sortedcodes,
+        EXCHANGES[sortedcodes],
+        session=session,
+        target_datetime=target_datetime,
+        logger=logger,
+    )
     for point in exchange:
-        point.update({
-            'sortedZoneKeys': point.pop('zoneKey'),
-            'netFlow': point.pop('value'),
-        })
+        point.update(
+            {
+                "sortedZoneKeys": point.pop("zoneKey"),
+                "netFlow": point.pop("value"),
+            }
+        )
         if sortedcodes in REVERSE_EXCHANGES:
-            point['netFlow'] = -point['netFlow']
+            point["netFlow"] = -point["netFlow"]
 
     return exchange
 
 
-def _fetch_series(zone_key, series_id, session=None, target_datetime=None,
-                  logger=None):
+def _fetch_series(zone_key, series_id, session=None, target_datetime=None, logger=None):
     """Fetches and converts a data series."""
 
     s = session or requests.Session()
 
     # local import to avoid the exception that happens if EIAPY token is not set
     # even if this module is unused
     from eiapy import Series
+
     series = Series(series_id=series_id, session=s)
 
     if target_datetime:
-        utc = tz.gettz('UTC')
-        #eia currently only accepts utc timestamps in the form YYYYMMDDTHHZ
-        end = target_datetime.astimezone(utc).strftime('%Y%m%dT%HZ')
-        start = (target_datetime.astimezone(utc) - datetime.timedelta(days=1)).strftime('%Y%m%dT%HZ')
+        utc = tz.gettz("UTC")
+        # eia currently only accepts utc timestamps in the form YYYYMMDDTHHZ
+        end = target_datetime.astimezone(utc).strftime("%Y%m%dT%HZ")
+        start = (target_datetime.astimezone(utc) - datetime.timedelta(days=1)).strftime(
+            "%Y%m%dT%HZ"
+        )
         raw_data = series.get_data(start=start, end=end)
     else:
         # Get the last 24 hours available.
         raw_data = series.last(24)
 
+    eia_error_message = None
+    # check nested key with forgiveness
+    try:
+        eia_error_message = raw_data["data"]["error"]
+    except KeyError:
+        pass
+    if eia_error_message:
+        logger.error(f"EIA error: {eia_error_message}")
+        return []
```

Comment:
Hmm, couldn't you do it like this instead?

```python
    try:
        eia_error_message = raw_data["data"]["error"]
        logger.error(f"EIA error: {eia_error_message}")
        return []
    except KeyError:
        pass
```

Or maybe even this? Not sure what's clearer:
```python
    eia_error_message = raw_data.get('data', {}).get('error')
    if eia_error_message: 
        logger.error(f"EIA error: {eia_error_message}")
        return []


```

Reply:
Good suggestion! Didn't know about the `.get` trick ;)